sources/src/ChatGPTPopover.tsx

```diff
@@ -1,79 +1,141 @@
+
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [open, isMinimized, handleMinimize]);
+
   return (
-    <Popover
-      open={open}
-      onClose={onClose}
-      keepMounted={false}
-      anchorReference="none"
-      anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
-      anchorPosition={{ top: 100, left: 100 }}
-      BackdropProps={{ invisible: false, sx: { backgroundColor: 'rgba(0, 0, 0, 0.5)' } }}
-      {...popoverProps}
-      sx={{
-        [`> .${paperClasses.root}`]: {
-          width,
-          height,
-          display: 'flex',
-          flexDirection: 'column',
-          bottom: 10,
-          right: 10
-        },
-        ...popoverProps?.sx
-      }}
-    >
-      <AppBar
-        position="static"
-        {...appBarProps}
+    <>
+      <Popover
+        open={open && !isMinimized}
+        onClose={onClose}
+        disableEscapeKeyDown={true}
+        keepMounted={isMinimized}
+        anchorReference="none"
+        anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
+        anchorPosition={{ top: 100, left: 100 }}
+        BackdropProps={{
+          invisible: false,
+          sx: { backgroundColor: 'rgba(0, 0, 0, 0.5)' },
+          onClick: (event) => {
+            event.stopPropagation();
+            setOpenAlertDialog(true);
+          }
+        }}
+        {...popoverProps}
         sx={{
-          display: 'flex',
-          flexDirection: 'row',
-          placeContent: 'space-between',
-          alignItems: 'center',
-          pl: 2,
-          pr: 1,
-          py: 1,
-          ...appBarProps?.sx
+          zIndex: theme.zIndex.modal + 1,
+          [`> .${paperClasses.root}`]: {
+            width,
+            height,
+            display: 'flex',
+            flexDirection: 'column',
+            bottom: 10,
+            right: 10
+          },
+          ...popoverProps?.sx
         }}
       >
-        <Typography variant="h6" color="inherit" component="div" children={appBarTitle} />
-        <IconButton color="inherit" aria-label="close" onClick={(e) => onClose?.(e, 'closeButton' as 'backdropClick')}>
-          <CloseRounded />
-        </IconButton>
-      </AppBar>
-      <ChatGPT {...chatGPTProps} sxs={{ root: { height: 'calc(100% - 56px)' }, ...chatGPTProps?.sxs }} />
-    </Popover>
+        <DialogHeader
+          title={appBarTitle}
+          onMinimizeButtonClick={handleMinimize}
+          onCloseButtonClick={(e) => onClose(e, null)}
+        />
+        <ChatGPT {...chatGPTProps} sxs={{ root: { height: 'calc(100% - 56px)' }, ...chatGPTProps?.sxs }} />
+      </Popover>
+      <MinimizedBar open={isMinimized} onMaximize={onMaximize} title={appBarTitle} />
+      <AlertDialog
+        sxs={{
+          root: {
+            zIndex: theme.zIndex.modal + 2
+          }
+        }}
+        disableBackdropClick
+        disableEscapeKeyDown
+        open={openAlertDialog}
+        title="Confirm closing this chat?"
```

Comment:
I guess we could simplify the language
```suggestion
        title="Close this chat?"
```

Reply:
Updated